### DISCUSSION TITLE
Ansys icepak coding error

### DISCUSSION THREAD

**Discussion by PowerbrainAI** [ansys/pyaedt#3901] [Q&A] [answered]
from pyaedt import Icepak, generate_unique_folder_name, downloads, settings
import os
from pyaedt import settings
import win32api
win32api.FormatMessage(-2147352565)
settings.use_grpc_api = False

temp_folder = generate_unique_folder_name()
package_temp_name, qfp_temp_name = downloads.download_icepak_3d_component(temp_folder)
print(temp_folder)

ipk = Icepak(projectname=os.path.abspath(os.path.join(temp_folder, "Heatsink.aedt")), specified_version="2022.1", non_graphical=True,
             close_on_exit=True, new_desktop_session=True)

ipk.modeler.get_object_from_name("Region").delete()

hs_base = ipk.modeler.create_box(position=[0, 0, 0], dimensions_list=[37.5, 37.5, 2], name="HS_Base")
hs_base.material_name="Al-Extruded"
hs_fin = ipk.modeler.create_box(position=[0, 0, 2], dimensions_list=[37.5, 1, 18], name="HS_Fin1")
hs_fin.material_name="Al-Extruded"
hs_fin.duplicate_along_line([0, 3.65, 0], nclones=11)

ipk.plot(show=False, export_path=os.path.abspath(os.path.join(temp_folder, "Heatsink.jpg")))

mesh_box = ipk.modeler.create_box(position=[-2, -2, -3], dimensions_list=[41.5, 41.5, 24])
mesh_box.model = False
mesh_region = ipk.mesh.assign_mesh_region([mesh_box.name])
mesh_region.UserSpecifiedSettings = True
mesh_region.MaxElementSizeX = "5mm"
mesh_region.MaxElementSizeY = "5mm"
mesh_region.MaxElementSizeZ = "1mm"
mesh_region.MinElementsInGap = "4"
mesh_region.MaxLevels = "2"
mesh_region.BufferLayers = "1"
mesh_region.update()

hs_fin5_object = ipk.modeler.get_object_from_name("HS_Fin1_5")
point_monitor_position = [0.5 * (hs_base.bounding_box[i] + hs_base.bounding_box[i + 3]) for i in range(2)] + [
    hs_fin5_object.bounding_box[-1]]  # average x,y, top z

ipk.monitor.assign_point_monitor(point_monitor_position, monitor_quantity=["Temperature", "HeatFlux"],
                                 monitor_name="TopPoint")
ipk.monitor.assign_face_monitor(hs_base.bottom_face_z.id, monitor_quantity="Temperature", monitor_name="Bottom")
ipk.monitor.assign_point_monitor_in_object("HS_Fin1_5", monitor_quantity="Temperature", monitor_name="Fin5Center")

os.mkdir(os.path.abspath(os.path.join(temp_folder, "componentLibrary")))
ipk.modeler.create_3dcomponent(
    os.path.abspath(os.path.join(temp_folder, "componentLibrary", "Heatsink.a3dcomp")),
    component_name="Heatsink",
    auxiliary_dict=True
)
ipk.close_project(save_project=False)
###################################################################################################
I ran the above code but I have this this problem. 
####################################################################################################
Traceback (most recent call last):
  File "D:\孔嘉泽\heatsinkmodel\heatsink.py", line 45, in <module>
    ipk.monitor.assign_point_monitor(point_monitor_position, monitor_quantity=["Temperature", "HeatFlux"],
  File "D:\Anaconda3\envs\kong\lib\site-packages\pyaedt\icepak.py", line 210, in monitor
    self._monitor._delete_removed_monitors()  # force update. some operations may delete monitors
  File "D:\Anaconda3\envs\kong\lib\site-packages\pyaedt\modules\monitor_icepak.py", line 592, in _delete_removed_monitors
    existing_monitors = self._app.odesign.GetChildObject("Monitor").GetChildNames()
  File "<COMObject InsertDesign>", line 2, in GetChildObject
  File "D:\Anaconda3\envs\kong\lib\site-packages\win32com\client\dynamic.py", line 368, in _ApplyTypes_
    result = self._oleobj_.InvokeTypes(
pywintypes.com_error: (-2147352567, '发生意外。', (0, None, None, None, 0, -2147024382), None)
####################################################################################################
Does anyone know what is going on?

**Comment by Samuelopez-ansys** [comment, marked as answer]
@PowerbrainAI New PyAEDT versions are not supported in 2022.1.

https://aedt.docs.pyansys.com/version/stable/Getting_started/About.html#dependencies

Please update the AEDT version, I suggest installing 2023.2. New PyAEDT releases are tested in the latest official release.

Thanks
